classes/payment-processor.php

```diff
@@ -88,6 +89,32 @@ public static function process( $order ) {
 		do_action( 'zco_after_process_payment', $payment, $order, $response );
 	}
 
+
+	/**
+	 * Check if the payment can transition from the current Zaver status to the new status.
+	 *
+	 * @docs https://api-docs-9sv4lksrb24zkgpv.zaver.com/v2/pay-link.html#payment-status
+	 *
+	 * @param string $new_status The new status to transition to.
+	 * @param string $current_status The current status of the payment.
+	 * @return bool True if the transition is allowed, false otherwise.
+	 */
+	public static function can_transition( $new_status, $current_status ) {
+		$new_status     = strtolower( $new_status );
+		$current_status = strtolower( $current_status );
+
+		if ( $current_status === 'created' && in_array( $new_status, array( 'cancelled', 'pending_merchant_capture', 'pending_confirmation', 'settled' ), true ) ) {
+			return true;
+		}
+
+		// While 'pending_merchant_capture' can transition to itself, we will not allow it so as to not perform the same processing again.
+		if ( $current_status === 'pending_merchant_capture' && in_array( $new_status, array( 'settled' ), true ) ) {
+			return true;
+		}
```

Comment:
Issue: Is it not possible for a `pending_merchant_capture` to for example get the status cancelled? Or for a `pending_confirmation` status to get confirmed or a cancel status later? I think this right now stops some possible flows that are currently working.

If the issue was that we could get the status `pending_merchant_capture` multiple times, or other statuses multiple times would it not be better to set what status we have handled as a metadata for the order when we do that, and check if the metadata has been set for that specific status instead? That should prevent the order notes from being repeated, and also allow any other status transition to go through even going forward.

Reply:
That sounds like a good solution.